src/passes/Memory64Lowering.cpp

```diff
@@ -79,9 +79,31 @@ struct Memory64Lowering : public WalkerPass<PostWalker<Memory64Lowering>> {
     if (memory->is64()) {
       wrapAddress64(curr->delta, curr->memory);
       auto* size = static_cast<Expression*>(curr);
-      extendAddress64(size, curr->memory);
+      // MemoryGrow returns -1 in the case of failure
+      // so we need to handle that
+      // local.set $tmp memory.grow X
+      // local.get
+      // i32.const -1
+      // if i32.eq
+      //   i64.const -1
+      // else
+      //   local.get
+      //   i32.extend
```

Comment:
```suggestion
      // (local.set $tmp (memory.grow X))
      // (if
      //   (i32.eq
      //     (local.get $tmp)
      //     (i32.const -1)
      //   )
      //   (then
      //     (i64.const -1)
      //   )
      //   (else
      //     (i32.extend
      //       (local.get $tmp)
      //     )
      //   )
      // )
```
(might as well describe it in structured form, as that is what the code below emits)